src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -133,7 +133,7 @@ private function replaceConstructor(ClassMethod $method)
             // Hydrate closures
             $bodyParts[] = "\$this->hydrateCallbacks[] = \\Closure::bind(function (\$object, \$values) {";
             foreach ($propertyNames as $propertyName) {
-                $bodyParts[] = "    if (isset(\$values['" . $propertyName . "'])) {";
+                $bodyParts[] = "    if (isset(\$values['$propertyName']) || \$object->$propertyName !== null && array_key_exists('$propertyName', \$values)) {";
```

Comment:
I will remove string interpolation for consistency sake, but you should think about using it since this is perfect use case for it (heavy use of string concatenation with simple variables only), as opposed to splitting strings all over with `".` Think about it as template system. IMO easier to read, because all you need to have in mind is that escaped variables are literal, otherwise they are interpolated. No longer need to break the flow of the code, no longer necessity to check if variable is in context of string or outside it (because it isn't visually different), no longer necessity to tune `'" . ` out of your mind when visualizing the output of the result :)